src/pages/index.tsx

```diff
@@ -34,8 +36,25 @@ export const getServerSideProps: GetServerSideProps = scaffold(
         await applySession(req, res);
 
         const reqWithSession = req as { session?: AppSession };
+
+        try {
+          const userRes = await context.z.resource('users', 'me').get();
+          context.user = userRes.data.data as ZetkinUser;
+        } catch (error) {
+          context.user = null;
+        }
+
         if (reqWithSession.session) {
           reqWithSession.session.tokenData = context.z.getTokenData();
+          if (context.user) {
+            try {
+              reqWithSession.session.organizations = await getOrganizations(
+                context
+              );
+            } catch (error) {
+              reqWithSession.session.organizations = null;
+            }
+          }
```

Comment:
I'm not sure why this code is needed on the index page, doesn't scaffold handle this stuff?

Reply:
Hm I'm not sure I know enough about the code, but I just put a console log on `index` and it never showed. So I think this code only has to do with the index page. But I don't know well enough to say for sure, curious what @richardolsson knows

Reply:
Thanks for tagging me!

I haven't really looked closely at any code in this PR, but @ziggabyte is right about architecture/flow. All users are always redirected to the index page after logging in. The index page is configured on the OAuth2 client as the redirect page, and is responsible for exchanging the authorization code for a token, and storing that in the session. Then the user may or may not be redirect to another page on the site. So this is always where the session is first created, and hence where the user's memberships/organizations is first retrieved.

For most users, the organizations will never be retrieved again. They're cached in the session data for the duration of their session. The only time `scaffold()` will retrieve organizations is when a logged in user tries to navigate to an organization to which they don't have access, at which point `scaffold()` will re-fetch the organizations to make sure that hasn't changed during the session. That could happen when someone logs in, notices they don't have access to organization X, tells a comrade who promotes them, and then tries again without logging out and back in again.